fendermint/app/src/app.rs

```diff
@@ -407,7 +409,7 @@ where
         Message = Vec<u8>,
         BeginOutput = FvmApplyRet,
         DeliverOutput = BytesMessageApplyRes,
-        EndOutput = PowerUpdates,
+        EndOutput = FvmEndOutput,
```

Comment:
Because we aren't doing anything with it, it's actually a needless constraint from the `App` to demand that it's the `end_block` where upgrades are applied, and not the `begin_block` for example. I can't think of a reason to prefer one over the other, so we can leave it at the discretion of the interpreter I think. 

Sorry, I know it was my idea.

Reply:
No problem, I'll change it back and call `update_app_version` inside the interpreter on upgrades